src/bridges/tokemak/DepositBridge.sol

```diff
@@ -0,0 +1,337 @@
+    uint256 inputValue = pendingInteractions[interactionNonce].inputValue;
+    require(inputValue > 0);
+        
+    address tAsset = inputAssetA.erc20Address;
+    require(tAsset != address(0),"Tokemak DepositBridge: INVALID_INPUT");
+    bool isAsync = true;
+    // Withdraw pending withdrawals
+    (outputValueA, isAsync) =  _finaliseWithdraw(tAsset, inputValue, interactionNonce);
+    interactionCompleted = !isAsync;
+  }
+
+    /**
+     * @dev Function to attempt finalising of as many interactions as possible within the specified gas limit
+     * Continue checking for and finalising interactions until we expend the available gas
+     * @param gasFloor The amount of gas that needs to remain after this call has completed
+     */
+    function finalisePendingInteractions(uint256 gasFloor) internal {
+        // check and finalise interactions until we don't have enough gas left to reliably update our state without risk of reverting the entire transaction
+        // gas left must be enough for check for next expiry, finalise and leave this function without breaching gasFloor
+        uint256 gasLoopCondition = MIN_GAS_FOR_CHECK_AND_FINALISE + MIN_GAS_FOR_FUNCTION_COMPLETION + gasFloor;
+        uint256 ourGasFloor = MIN_GAS_FOR_FUNCTION_COMPLETION + gasFloor;
+        while (gasleft() > gasLoopCondition) {
+            // check the heap to see if we can finalise an expired transaction
+            // we provide a gas floor to the function which will enable us to leave this function without breaching our gasFloor
+            (bool available, uint256 nonce) = checkForNextInteractionToFinalise(ourGasFloor);
+            if (!available) {
+                break;
+            }
+            // make sure we will have at least ourGasFloor gas after the finalise in order to exit this function
+            uint256 gasRemaining = gasleft();
+            if (gasRemaining <= ourGasFloor) {
+                break;
+            }
+            uint256 gasForFinalise = gasRemaining - ourGasFloor;
+            // make the call to finalise the interaction with the gas limit        
+            try IRollupProcessor(rollupProcessor).processAsyncDefiInteraction{gas: gasForFinalise}(nonce) returns (bool interactionCompleted) {
+                // no need to do anything here, we just need to know that the call didn't throw
+            } catch {
+                break;
+            }
+        }
+    }
+  
+    /**
+     * @dev Function to get the next interaction to finalise
+     * @param gasFloor The amount of gas that needs to remain after this call has completed
+     */
+    function checkForNextInteractionToFinalise(uint256 gasFloor)
+        internal
+        returns (
+            bool ,
+            uint256 
+        )
+    {
+        // do we have any expiries and if so is the earliest expiry now expired
+        if (nonces.length == 0) {
+            return (false, 0);
+        }
+       
+        uint256 minGasForLoop = gasFloor;
```

Comment:
I think you will need more gas than gasFloor for this loop won't you? I think convert will be in danger of reverting due to out of gas.